File: CS/WebSite/App_Code/ImportExportCommands.cs
using System.IO;
using DevExpress.Web.ASPxClasses;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler.iCalendar;
using DevExpress.XtraScheduler;

public class ExportAppointmentCallbackCommand : MenuViewCallbackCommand {
    public ExportAppointmentCallbackCommand(ASPxScheduler control)
        : base(control) {
    }

    protected override void ExecuteCore() {
        //PostCalendarFile(Control.SelectedAppointments);
        PostCalendarFile(Control.Storage.Appointments.Items);
    }

    void PostCalendarFile(AppointmentBaseCollection appointments) {
        iCalendarExporter exporter = new iCalendarExporter(Control.Storage, appointments);
        
        if (appointments.Count == 0)
            return;
        
        MemoryStream memoryStream = new MemoryStream();

        exporter.Export(memoryStream);
        memoryStream.WriteTo(Control.Page.Response.OutputStream);
        Control.Page.Response.ContentType = "text/calendar";
        Control.Page.Response.AddHeader("Content-Disposition", "attachment; filename=appointment.ics");
        Control.Page.Response.End();
    }
}

public class ImportAppointmentsCallbackCommand : MenuViewCallbackCommand {
    public ImportAppointmentsCallbackCommand(ASPxScheduler control)
        : base(control) {
    }

    protected override void ExecuteCore() {
        Stream stream = GetStream();

        if (stream != null) {
            Control.Storage.Appointments.Clear();

            iCalendarImporter importer = new iCalendarImporter(Control.Storage);
            importer.Import(stream);
        }
    }

    private Stream GetStream() {
        byte[] buf = Control.Page.Session["UploadedFile"] as byte[];
        Stream stream = new MemoryStream(buf);
        return stream;
    }
}

File: VB/WebSite/App_Code/ImportExportCommands.vb
Imports System.IO
Imports DevExpress.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler.iCalendar
Imports DevExpress.XtraScheduler

Public Class ExportAppointmentCallbackCommand
    Inherits MenuViewCallbackCommand

    Public Sub New(ByVal control As ASPxScheduler)
        MyBase.New(control)
    End Sub

    Protected Overrides Sub ExecuteCore()
        'PostCalendarFile(Control.SelectedAppointments);
        PostCalendarFile(Control.Storage.Appointments.Items)
    End Sub

    Private Sub PostCalendarFile(ByVal appointments As AppointmentBaseCollection)
        Dim exporter As New iCalendarExporter(Control.Storage, appointments)

        If appointments.Count = 0 Then
            Return
        End If

        Dim memoryStream As New MemoryStream()

        exporter.Export(memoryStream)
        memoryStream.WriteTo(Control.Page.Response.OutputStream)
        Control.Page.Response.ContentType = "text/calendar"
        Control.Page.Response.AddHeader("Content-Disposition", "attachment; filename=appointment.ics")
        Control.Page.Response.End()
    End Sub
End Class

Public Class ImportAppointmentsCallbackCommand
    Inherits MenuViewCallbackCommand

    Public Sub New(ByVal control As ASPxScheduler)
        MyBase.New(control)
    End Sub

    Protected Overrides Sub ExecuteCore()
        Dim stream As Stream = GetStream()

        If stream IsNot Nothing Then
            Control.Storage.Appointments.Clear()

            Dim importer As New iCalendarImporter(Control.Storage)
            importer.Import(stream)
        End If
    End Sub

    Private Function GetStream() As Stream
        Dim buf() As Byte = TryCast(Control.Page.Session("UploadedFile"), Byte())
        Dim stream As Stream = New MemoryStream(buf)
        Return stream
    End Function
End Class